metadata-io/src/main/java/com/linkedin/metadata/service/UpdateIndicesService.java

```diff
@@ -111,10 +114,25 @@ public UpdateIndicesService(
 
   public void handleChangeEvent(@Nonnull final MetadataChangeLog event) {
     try {
-      if (UPDATE_CHANGE_TYPES.contains(event.getChangeType())) {
-        handleUpdateChangeEvent(event);
-      } else if (event.getChangeType() == ChangeType.DELETE) {
-        handleDeleteChangeEvent(event);
+      MCLBatchItemImpl batch =
+          MCLBatchItemImpl.builder().build(event, _entityRegistry, systemEntityClient);
+
+      Stream<MCLBatchItem> sideEffects =
+          _entityRegistry
+              .getMCLSideEffects(
+                  event.getChangeType(), event.getEntityType(), event.getAspectName())
+              .stream()
+              .flatMap(
+                  mclSideEffect ->
+                      mclSideEffect.apply(List.of(batch), _entityRegistry, systemEntityClient));
+
+      for (MCLBatchItem mclBatchItem : Stream.concat(Stream.of(batch), sideEffects).toList()) {
```

Comment:
Batching together MCLs is a significant change, have we considered all of the cases where parallel batches contain conflicting changes? I feel like it would be easy to get ourselves in a deadlock scenario from the Kafka side with this i.e. multiple consumers processing multiple batches with the same aspect being modified could repeatedly fail batches without making progress and having to reprocess the entire batch, with only one MCL processed (current state) this can still happen (and has), but one of the consumers is guaranteed to make progress whereas with a batch this is not the case. I'm not seeing rollback logic in here so it seems like the entire batch gets reprocessed.

Reply:
The batches contain default aspects today and they do conflict sometimes. This is handled by retry logic in the elasticsearch processor (and transaction retries in the ebean MCP/mce stage). We have deadlock recovery logic already. The batching in this PR is extending it to include those default aspects per normal as well as  possible side effect aspects. If there is a deadlock and failure it would be re-tried either at the driver level (es client or ebean client) and then for mcl/mae consumer the message would be reprocessed.